src/librbd/image/RemoveRequest.cc

```diff
@@ -58,6 +58,8 @@ Context *RemoveRequest<I>::handle_open_image(int *result) {
 
   if (*result < 0) {
     lderr(m_cct) << "error opening image: " << cpp_strerror(*result) << dendl;
+    if (*result != -ENOENT)
```

Comment:
Squash this down into the original commit. Still need to switch contexts I believe (and delete the m_image_ctx)